File: 252_meeting_rooms.js
/**
 * @param {number[][]} intervals
 * @return {boolean}
 */
var canAttendMeetings = function(intervals) {
    if (intervals.length < 2){
        return true;
    }
    const sorted = [...intervals].sort((a,b) => a[0]-b[0]);
    let currentEnd = sorted[0][1];
    for (let i = 1; i < sorted.length; i++){
        if (currentEnd > sorted[i][0]) return false;
        if (currentEnd < sorted[i][1]) currentEnd = sorted[i][1];
    }
    return true;
};
//Time Complexity O(n log n), Space Complexity O(1);

File: 252_meeting_rooms.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        if len(intervals) < 2:
            return True
        
        sortedIntervals = sorted(intervals, key=lambda x: x[0])
        currentEnd = sortedIntervals[0][1]

        for i in range(1, len(sortedIntervals)):
            if currentEnd > sortedIntervals[i][0]:
                return False
            if currentEnd < sortedIntervals[i][1]:
                currentEnd = sortedIntervals[i][1]
        return True